x-pack/plugin/migrate/src/main/java/org/elasticsearch/xpack/migrate/action/CopyLifecycleIndexMetadataTransportAction.java

```diff
@@ -64,7 +63,8 @@ public CopyLifecycleIndexMetadataTransportAction(
         this.executor = new SimpleBatchedAckListenerTaskExecutor<>() {
             @Override
             public Tuple<ClusterState, ClusterStateAckListener> executeTask(UpdateIndexMetadataTask task, ClusterState clusterState) {
-                return new Tuple<>(applyUpdate(clusterState, task), task);
+                final ProjectMetadata projectMetadata = projectResolver.getProjectMetadata(clusterState);
```

Comment:
This is an example that would fail in a context where multiple projects are doing things concurrently, but it won't fail in the tests we currently have. By using the project resolver here, we depend on the thread context, but the thread context is not updated for every task in a batch update. Therefore, we'll need to pass the project ID to the `UpdateIndexMetadataTask` when we create it.

Reply:
I think I've addressed this via 7d17f7a0ea6e464fe37012e2259acb35bce091a9, but please confirm that this is what you're looking for.